fastddsspy_participants/test/blackbox/model/EndpointDatabaseTest.cpp

```diff
@@ -0,0 +1,418 @@
+            std::make_shared<test::SpyDdsParticipantMock>(id_);
+
+    // Create DDS Pipe
+    std::shared_ptr<DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->endpoint_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send messages
+    spy::participants::EndpointInfoData new_data;
+    random_endpoint_info(new_data, false);
+    dds_reader->simulate_data_reception(
+        std::make_unique<spy::participants::EndpointInfoData>(new_data));
+
+    // Wait
+    std::this_thread::sleep_for(std::chrono::milliseconds(test::WAIT_MS));
+
+    // Check information
+    ASSERT_FALSE(model->endpoint_database_.at(new_data.info.guid).active);
+}
+
+TEST(EndpointDatabaseTest, is_writer)
+{
+    // Create Model
+    std::shared_ptr<spy::participants::SpyModel> model =
+            std::make_shared<spy::participants::SpyModel>();
+
+    const types::ParticipantId id_ = "DDS Spy mock participant";
+
+    std::shared_ptr<test::SpyDdsParticipantMock> dds_participant =
+            std::make_shared<test::SpyDdsParticipantMock>(id_);
+
+    // Create DDS Pipe
+    std::shared_ptr<DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->endpoint_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send messages
+    spy::participants::EndpointInfoData new_data;
+    random_endpoint_info(new_data, true, 1, ddspipe::core::types::EndpointKind::writer);
+    dds_reader->simulate_data_reception(
+        std::make_unique<spy::participants::EndpointInfoData>(new_data));
+
+    // Wait
+    std::this_thread::sleep_for(std::chrono::milliseconds(test::WAIT_MS));
+
+    // Check information
+    ASSERT_EQ(model->endpoint_database_.at(new_data.info.guid).kind, ddspipe::core::types::EndpointKind::writer);
+}
+
+TEST(EndpointDatabaseTest, is_reader)
+{
+    // Create Model
+    std::shared_ptr<spy::participants::SpyModel> model =
+            std::make_shared<spy::participants::SpyModel>();
+
+    const types::ParticipantId id_ = "DDS Spy mock participant";
+
+    std::shared_ptr<test::SpyDdsParticipantMock> dds_participant =
+            std::make_shared<test::SpyDdsParticipantMock>(id_);
+
+    // Create DDS Pipe
+    std::shared_ptr<DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->endpoint_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send messages
+    spy::participants::EndpointInfoData new_data;
+    random_endpoint_info(new_data, true, 1, ddspipe::core::types::EndpointKind::reader);
+    dds_reader->simulate_data_reception(
+        std::make_unique<spy::participants::EndpointInfoData>(new_data));
+
+    // Wait
+    std::this_thread::sleep_for(std::chrono::milliseconds(test::WAIT_MS));
+
+    // Check information
+    ASSERT_EQ(model->endpoint_database_.at(new_data.info.guid).kind, ddspipe::core::types::EndpointKind::reader);
+}
+
```

Comment:
Please, add a test that modifies the active/inactive variable of an already existing endpoint